packages/progress-circle/src/ProgressCircle.base.ts

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright 2025 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {
+    PropertyValues,
+    SizedMixin,
+    SpectrumElement,
+} from '@spectrum-web-components/base';
+import {
+    property,
+    query,
+} from '@spectrum-web-components/base/src/decorators.js';
+import { getLabelFromSlot } from '@spectrum-web-components/shared/src/get-label-from-slot.js';
+
+export class ProgressCircleBase extends SizedMixin(SpectrumElement, {
```

Comment:
I think we could declare this as an abstract class:
```suggestion
export abstract class ProgressCircleBase extends SizedMixin(SpectrumElement, {
```